File: src/reflection/reflective_runtime/attribute.cpp
#include "attribute.h"

namespace reflective {

attribute::attribute(const std::string& name)
    : name_(name)
{
}

attribute::attribute(const std::string& name, const key_values_type& key_values)
    : name_(name), key_values_(key_values)
{
}

attribute::~attribute()
{

}

}

File: dotnet/Centauri.SourceGenerators/Reflective/Attribute.cs
namespace Centauri.SourceGenerators.Reflective;

public class Attribute
{
    public string? RawAttribute { get; set; }

    public string? Name { get; set; }

    public IDictionary<string, string> Parameters { get; set; } = new Dictionary<string, string>();

    public bool Valid { get; set; }
}